Dockerfile

```diff
@@ -53,4 +53,7 @@ COPY bin/* ${PG_LIB}/bin/
 COPY ci/ ./ci/
 
 VOLUME ${PG_HOME}
+#-------------------------------------------------------------------------------
+# https://github.com/ufoscout/docker-compose-wait
+COPY wait-2.7.3 /wait
```

Comment:
There's a few issues, here; more about those in other sections. But for this, the main one is there's a vendoring of a specific version which will never be upgraded by either updating libs or base images, meaning no fixes will ever be applied, non-standard usage contrary to FHS (also I'll be honest, I'm less worried about this, and I myself break that sometimes), and no checksumming or such.

Reply:
It seems I can't add a comment to the binary file added (`wait-2.7.3`). Separate to my other points, there are some issues with this:
- I can't be sure where it's come from or about its integrity, since it's a binary blob without checksums matching anything elsewhere;
- in general, I think it's better to do a checksummed download within an image build, rather than add a binary to the repo itself, unless it's really absolutely necessary;
- it will likely never get updated, as mentioned;
- just as importantly, I'm concerned it would violate the licence; `wait` following your link lists it as Apache 2.0 licence, which has a licence and copyright notice obligation, which doesn't appear to be satisfied here. Off the top of my head, I'm also unsure about Apache 2.0-MIT compatibility; I *think* they're deemed compatible (if of course the licence requirements are satisfied), but it would be necessary to check.

Reply:
Ok. I agree with you.

So as I can understand we have to problems:

1. Update & checksum - I can replace file to next strings.

    ```Dockerfile
     ADD https://github.com/ufoscout/docker-compose-wait/releases/download/2.7.3/wait /wait
     RUN chmod +x /wait
     ```
2. License - as I understood:
    First one: If we don't change wait-service, we don't need to use this license as main.
    Second: We can add information about wait-service and Apache license to Readme file.

Reply:
I am not a lawyer, but I don't think that's necessarily the correct interpretation of the Apache licence. However, I agree it has no effect on main. Regardless, however, I've been thinking about this point further, too, and I feel it necessary to clarify: what problem, exactly, did you experience, leading to adding the wait script? Since as I mentioned, there is extensive healthchecking support in this project already, and even if containers start in the wrong order, they should usually fix themselves (after the bootstrapping process is finished, which is a specific exception for setting up the cluster). I'm currently not seeing enough justification to add the wait script, regardless of how it is installed or packaged or checksummed, since I'm unclear about what it gives us compared to what's already there.

I think this PR has become quite long, and seems to be be addressing multiple different things (e.g. Pgpool, wait script, minor fixes and extensions to documentation, additional example file for installing on Kubernetes via Docker Stack and kubectl, …). It might be easier to split this stuff apart into smaller PRs. However, I also don't want there to be wasted work, so to summarise what I understand from this PR so far:
- minor fixes and extensions to documentation are welcome (pretty much as-is);
- additional example file for installation Kubernetes via Docker Stack and kubectl also is fine, but note much of it is likely very similar to the files already in place for Docker Swarm via Docker Stack (after all, that's much of the point of the multi-orchestrator Docker Stack, providing an interface for this);
- I'm thinking that this project is likely not an appropriate place for adding Pgpool, for the reasons given above;
- I'm currently struggling to understand what issue there was with the healthchecks already present, motivating the adding of the wait script.
Hope that helps to clarify. Thanks again.